### PR TITLE
Added APIExceptionLoggerFilter.cs For Global NLog

### LINKED ISSUE #176: logger not created For Web ApI Controller

### PR DESCRIPTION
**Pull request by ankit213**
Fixed #176

Changed Files :  

1. **ApiExceptionLoggerFilter.cs** : added ApiExceptionLoggerFilter file(for webapi controller global logs).
2.**Global.asax.cs** :update Global.asax.cs file (ApiExceptionLoggerFilter added in  GlobalConfiguration.Configuration.Filters).

### CODE REVIEW

File: Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            _logger = logger;
+        }
+
+        public bool AllowMultiple { get { return true; } }
+
+        public Task ExecuteExceptionFilterAsync(
+                HttpActionExecutedContext actionExecutedContext,
+                CancellationToken cancellationToken)
+        {
+            return Task.Factory.StartNew(() =>
```
Comment (chintans) [inline Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs]: Why have you used task here?

**Review by chintans — CHANGES_REQUESTED** [review]

File: Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs (hunk 1 above)
Comment (ankit213) [reply Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs]: Removed the create task, now only return Task.FromResult(0) to display this task executed successfully.

File: Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        public bool AllowMultiple { get { return true; } }
+
+        public Task ExecuteExceptionFilterAsync(
+                HttpActionExecutedContext actionExecutedContext,
+                CancellationToken cancellationToken)
+        {
+            _logger.Error("Web Service Error Message:" + actionExecutedContext.Exception.Message);
+            _logger.Error("Web Service Error StackTrace:" + actionExecutedContext.Exception.StackTrace);
+            return Task.FromResult(0);
```
Comment (chintans) [inline Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs]: I still don't understand usage of Task.FromResult(0) ?

**Review by chintans — CHANGES_REQUESTED** [review]

File: Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs (hunk 2 above)
Comment (ankit213) [reply Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs]: IExceptionFilter has method "ExecuteExceptionFilterAsync" and its return type is task so I return task.

Comment (chintans) [reply Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs]: It is an async method and that is why it is returning Task. You have not implemented async method properly. Task type means the method does not return anything. Generic Task means method will return something.

Comment (ankit213) [reply Slack.Automation/Promact.Erp.Core/ActionFilters/ApiExceptionLoggerFilter.cs]: I have updated method, Now It's implemented async properly.

**Review by chintans — APPROVED** [review]

**Pull request merged**